benchmarks/util/schema_proto2_to_proto3_util.h

```diff
@@ -0,0 +1,138 @@
+      if (old_file->message_type(i)->options().message_set_wire_format()) {
+        file.mutable_message_type()->DeleteSubrange(i, 1);
+        continue;
+      }
+      StripMessage(old_file->message_type(i), file.mutable_message_type(i));
+    }
+    for (int i = file.mutable_extension()->size() - 1; i >= 0; i--) {
+      auto field = old_file->extension(i);
+      if (field->type() == FieldDescriptor::TYPE_GROUP
+          || IsMessageSet(field->message_type())
+          || IsMessageSet(field->containing_type())) {
+        file.mutable_extension()->DeleteSubrange(i, 1);
+      }
+    }
+  }
+
+ private:
+  static bool IsMessageSet(const Descriptor *descriptor) {
+    if (descriptor != nullptr
+        && descriptor->options().message_set_wire_format()) {
+      return true;
+    }
+    return false;
+  }
+
+  static void StripMessage(const Descriptor *old_message,
+                           DescriptorProto *new_message) {
+    for (int i = new_message->mutable_field()->size() - 1; i >= 0; i--) {
+       if (old_message->field(i)->type() == FieldDescriptor::TYPE_GROUP
+          || IsMessageSet(old_message->field(i)->message_type())) {
+        new_message->mutable_field()->DeleteSubrange(i, 1);
+       }
+    }
+    for (int i = new_message->mutable_extension()->size() - 1; i >= 0; i--) {
+      auto field_type_name = new_message->mutable_extension(i)->type_name();
+      if (old_message->extension(i)->type() == FieldDescriptor::TYPE_GROUP
+          || IsMessageSet(old_message->extension(i)->containing_type())
+          || IsMessageSet(old_message->extension(i)->message_type())) {
+        new_message->mutable_extension()->DeleteSubrange(i, 1);
+      }
+    }
+    for (int i = 0; i < new_message->mutable_nested_type()->size(); i++) {
+      StripMessage(old_message->nested_type(i),
+                   new_message->mutable_nested_type(i));
+    }
+  }
+
+};
+
+class SchemaAddZeroEnumValue {
+
+ public:
+  SchemaAddZeroEnumValue() : total_added_(0) {}
+
+  void ScrubEnum(EnumDescriptorProto *enum_type) {
+    if (enum_type->value(0).number() != 0) {
+      bool has_zero = false;
+      for (int j = 0; j < enum_type->value().size(); j++) {
+        if (enum_type->value(j).number() == 0) {
+          EnumValueDescriptorProto temp_enum_value;
+          temp_enum_value.CopyFrom(enum_type->value(j));
+          enum_type->mutable_value(j)->CopyFrom(enum_type->value(0));
+          enum_type->mutable_value(0)->CopyFrom(temp_enum_value);
+          has_zero = true;
+          break;
+        }
+      }
+      if (!has_zero) {
+        enum_type->mutable_value()->Add();
+        for (int i = enum_type->mutable_value()->size() - 1; i > 0; i--) {
+          enum_type->mutable_value(i)->CopyFrom(
+              *enum_type->mutable_value(i - 1));
+        }
+        enum_type->mutable_value(0)->set_number(0);
+        std::stringstream ss;
+        ss << "ADDED_ZERO_VALUE_" << total_added_++;
```

Comment:
You can use std::to_string(int) in C++11.

Reply:
Done